cluster-autoscaler/processors/processors.go

```diff
@@ -89,9 +89,10 @@ func DefaultProcessors(options config.AutoscalingOptions) *AutoscalingProcessors
 		}),
 		ScaleUpStatusProcessor: status.NewDefaultScaleUpStatusProcessor(),
 		ScaleDownNodeProcessor: nodes.NewPreFilteringScaleDownNodeProcessor(),
+		// TODO NewMaxNodesProcessor should be called after NewAtomicResizeFilteringProcessor
+		// to have maximum removable nodes returned by ScaleDownSetProcessor
 		ScaleDownSetProcessor: nodes.NewCompositeScaleDownSetProcessor(
 			[]nodes.ScaleDownSetProcessor{
-				nodes.NewMaxNodesProcessor(),
 				nodes.NewAtomicResizeFilteringProcessor(),
```

Comment:
If there's just one sub-processor here, no need to use `NewCompositeScaleDownSetProcessor`. Also, TODO doesn't apply anymore.